File: Data Structures & Algorithms/non-overlapping-intervals/submission-3.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:

        intervals.sort(key = lambda i: i[0])

        curr = 0
        res = 0

        for i in range(1, len(intervals)):
            if intervals[i][0] < intervals[curr][1]:
                res += 1
                if intervals[i][1] < intervals[curr][1]:
                    curr = i    
            else:
                curr = i
            
        return res

File: Data Structures & Algorithms/counting-bits/submission-3.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> res;
        int mask = 1;
        for (int i = 0; i <= n; i++){
            if (i == (mask << 1)){
                mask <<= 1;
            }
            
            if (i < 2){
                res.push_back(i);
            } else {
                cout << "i: " << i << " mask: " << mask << endl;
                res.push_back(res[i - mask] + 1);
            }
        }

        return res;
    }
};
